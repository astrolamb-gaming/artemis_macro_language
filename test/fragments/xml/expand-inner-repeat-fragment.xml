<mission_data version="1.0">
    <templates>
        <template name="hello">
            <big_message title="${message}" />
            <repeat _range="Test">
                <set_variable name="${foo}" value="${message}" />
            </repeat>
        </template>
        <template name="helloEvent">
            <event name="${message}">
            <big_message title="${message}" />
            <repeat _range="Test">
                <set_variable name="${foo}" value="${message}" />
            </repeat>
            </event>
        </template>
        <template name="helloEvent">
            <event name="${message}">
            <big_message title="${message}" />
            <repeat _range="Test">
                <set_variable name="${foo}" value="${message}" />
            </repeat>
            </event>
        </template>
    </templates>
    <values>
        <value OUTER="Hello, nested" />
        <value TEST="0" />
        <range name="Test">
            <value foo="bar" />
            <value foo="baz" />
        </range>
         <range name="Ships">
            <value foo="Art" />
            <value foo="Ray Tracer" />
        </range>
    </values>

    <expand>
        <start>
            <expand _template="hello" message="${OUTER}" />
        </start>
        <expand _template="helloEvent" message="${OUTER}" />
    </expand>
    <expand _template="helloEvent" message="${OUTER}" />
    <repeat _range="Ships">
        <expand _template="helloEvent" message="${foo}" />
    </repeat>
    <event name="Empty"/>
</mission_data>