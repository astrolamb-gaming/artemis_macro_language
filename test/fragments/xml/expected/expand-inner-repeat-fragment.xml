<mission_data version="1.0">
  <start>
    <big_message title="Hello, nested"/>
    <set_variable name="bar" value="Hello, nested"/>
    <set_variable name="baz" value="Hello, nested"/>
  </start>
  <event name="Hello, nested">
    <big_message title="Hello, nested"/>
    <set_variable name="bar" value="Hello, nested"/>
    <set_variable name="baz" value="Hello, nested"/>
  </event>
  <event name="Hello, nested">
    <big_message title="Hello, nested"/>
    <set_variable name="bar" value="Hello, nested"/>
    <set_variable name="baz" value="Hello, nested"/>
  </event>
  <event name="Art">
    <big_message title="Art"/>
    <set_variable name="bar" value="Art"/>
    <set_variable name="baz" value="Art"/>
  </event>
  <event name="Ray Tracer">
    <big_message title="Ray Tracer"/>
    <set_variable name="bar" value="Ray Tracer"/>
    <set_variable name="baz" value="Ray Tracer"/>
  </event>
</mission_data>